sfcsample/VDecAccelVA.cpp

```diff
@@ -0,0 +1,743 @@
+            m_vaIDs.push_back(vaID);
+    }
+
+    // Check if surfaces created is equal to requested.
+    if (m_vaIDs.size() != m_DecodeDesc.surfaces_num)
+    {
+        if(VAFAILED(vaStatus))
+            printf("Create surface fail\n");
+        return 1;
+    }
+
+    // Create context
+    vaStatus = vaCreateContext(
+        m_vaDisplay,
+        m_vaConfigID,
+        aligned_width,
+        aligned_height,
+        VA_PROGRESSIVE,
+        &(m_vaIDs.at(0)),
+        m_vaIDs.size(),
+        &m_vaContextID
+        );
+    if(VAFAILED(vaStatus))
+            printf("vaCreateContext fail\n");
+
+    check_process_pipeline_caps(m_DecodeDesc);
+
+    return vaStatus;
+}
+
+
+void mvaccel::VDecAccelVAImpl::Close()
+{
+    for (auto it = m_vaIDs.begin(); it != m_vaIDs.end(); ++it)
+        delete_surface(*it);
+
+    vaDestroyConfig(m_vaDisplay, m_vaConfigID);
+    vaDestroyContext(m_vaDisplay, m_vaContextID);
+
+    //m_locks.clear();
+    m_images.clear();
+    m_vaIDs.clear();
+    m_vaSurfAttribs.clear();
+
+    m_vaConfigID = 0;
+    m_vaContextID = 0;
+}
+
+
+uint32_t mvaccel::VDecAccelVAImpl::GetSurfaceID(uint32_t index)
+{
+    assert(index < m_vaIDs.size());
+    return m_vaIDs[index];
+}
+
+//////////////////////////////////////////////////////////////////////////
+// member functions
+//////////////////////////////////////////////////////////////////////////
+
+/**
+ * @brief   Check if video decode acceleration description is supported.
+ * @param   cc Video decode acceleration description.
+ * @return  true if supported, false if not.
+ */
+bool mvaccel::VDecAccelVAImpl::is_config_compatible(DecodeDesc& desc)
+{
+    if (!is_slice_mode_supported(desc))
+        return false;
+
+    if (!is_encryption_supported(desc))
+        return false;
+
+    if(!is_sfc_config_supported(desc))
+        return false;
+
+    return true;
+}
+
+/**
+ * @brief   Check if long or short format is supported not not.
+ * @param   cc Video decode acceleration description.
+ * @return  true if supported, false if not.
+ */
+bool mvaccel::VDecAccelVAImpl::is_slice_mode_supported(DecodeDesc& desc)
+{
+    VAConfigAttrib vaAttrib;
+    memset(&vaAttrib, 0, sizeof(vaAttrib));
+    vaAttrib.type = VAConfigAttribDecSliceMode;
+    vaAttrib.value = 0;
+
+    vaGetConfigAttributes(
+        m_vaDisplay,
+        m_vaProfile,
+        m_vaEntrypoint,
+        &vaAttrib,
+        1);
+
+        return true;
+}
+
+/**
+ * @brief   Check if encryption is supported or not.
+ * @param   cc Video decode acceleration description.
+ * @return  true if supported, false if not.
+ */
+bool mvaccel::VDecAccelVAImpl::is_encryption_supported(DecodeDesc& desc)
+{
+    VAConfigAttrib vaAttrib;
+    memset(&vaAttrib, 0, sizeof(vaAttrib));
+    vaAttrib.type = VAConfigAttribEncryption;
+    vaAttrib.value = 0;
+
+    vaGetConfigAttributes(
+        m_vaDisplay,
+        m_vaProfile,
+        m_vaEntrypoint,
+        &vaAttrib,
+        1
+        );
+
+    return true;
```

Comment:
Same with preceding function.

Reply:
add encryption attribute check in function is_encryption_supported